vmdb/spec/requests/api/vms_spec.rb

```diff
@@ -637,4 +643,72 @@ def gen_request_data(action, data, *hrefs)
       expect(vm2.reload.evm_owner).to eq(@user)
     end
   end
+
+  context "Vm add_event action" do
+    it "add_event to an invalid vm" do
+      update_user_role(@role, action_identifier(:vms, :add_event))
+      basic_authorize @cfme[:user], @cfme[:password]
+
+      @success = run_post("#{@cfme[:vms_url]}/999999", gen_request(:add_event))
+
+      expect(@success).to be_false
+      expect(@code).to eq(404)
+    end
+
+    it "add_event to an invalid vm without appropriate role" do
+      basic_authorize @cfme[:user], @cfme[:password]
+
+      @success = run_post("#{@cfme[:vms_url]}/999999", gen_request(:add_event))
+
+      expect(@success).to be_false
+      expect(@code).to eq(403)
+    end
+
+    it "add_event to a single Vm" do
+      update_user_role(@role, collection_action_identifier(:vms, :add_event))
+      basic_authorize @cfme[:user], @cfme[:password]
+
+      vm = FactoryGirl.create(:vm_vmware, :host => @host, :ems_id => @ems.id, :power_state => "on")
+      vm_url = "#{@cfme[:vms_url]}/#{vm.id}"
+
+      @success = run_post(vm_url, gen_request_data(:add_event, :event_type => "special", :event_message => "message"))
+
+      expect(@success).to be_true
+      expect(@code).to eq(200)
+      expect(@result).to have_key("success")
+      expect(@result["success"]).to be_true
+      expect(@result).to have_key("message")
+      expect(@result["message"]).to match(/adding event/i)
+      expect(@result).to have_key("href")
+      expect(@result["href"]).to match(vm_url)
+    end
+
+    it "add_event to multiple Vms" do
+      update_user_role(@role, collection_action_identifier(:vms, :add_event))
+      basic_authorize @cfme[:user], @cfme[:password]
+
+      vm1 = FactoryGirl.create(:vm_vmware, :host => @host, :ems_id => @ems.id, :power_state => "on")
+      vm2 = FactoryGirl.create(:vm_vmware, :host => @host, :ems_id => @ems.id, :power_state => "on")
```

Comment:
well, tests didn't fail.  But you are right, I'm looking at the current rspecs for vm start, stop, I set raw_power_state to "powerOn".  I'll update here accordingly. Thanks.